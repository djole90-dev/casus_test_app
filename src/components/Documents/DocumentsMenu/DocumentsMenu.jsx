import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types'
import { Collapse, Box } from '@material-ui/core';
import {
  Container,
  Heading,
  ArrowDropDownIcon,
  ArrowRightIcon,
  ListContainer,
  MenuItem,
  Text
} from './DocumentsMenu.styles';

export const ListItem = ({ text, active, name, onClick }) => (
  <MenuItem onClick={onClick}>
    <Text active={active} data-categoryname={name}>
      {text}
    </Text>
  </MenuItem>
);


export const DocumentsMenu = ({ categories, filterDocumentsByCategory }) => {
  const [ isOpen, setCollaps ] = useState(true);
  const [ selectedCategory, setSelectedCategory ] = useState(null);

  const handleListItemClick = (e) => {
    setSelectedCategory(e.target.dataset.categoryname);
  };

  const handleArrowClick = () => setCollaps(prev => !prev)

  useEffect(() => {
    filterDocumentsByCategory(selectedCategory)
  }, [selectedCategory, filterDocumentsByCategory])

  return (
    <Container>
      <Box onClick={() => setSelectedCategory(null)}>
        <Heading active={!selectedCategory ? 'true' : null}>
          {isOpen ? (
            <ArrowDropDownIcon onClick={handleArrowClick} />
          ) : (
            <ArrowRightIcon onClick={handleArrowClick} />
          )}CASUS vorlagen
        </Heading>
      </Box>
      <Collapse in={isOpen}>
        <ListContainer>
          {categories.map((cat) => (
            <ListItem
              onClick={handleListItemClick}
              text={cat}
              name={cat}
              key={cat}
              active={cat === selectedCategory ? 'true' : null}
            />
          ))}
        </ListContainer>
      </Collapse>
    </Container>
  );
};

ListItem.propTypes = {
  text: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired
}

DocumentsMenu.propTypes = {
  categories: PropTypes.array.isRequired,
  filterDocumentsByCategory: PropTypes.func.isRequired
}


export default DocumentsMenu
