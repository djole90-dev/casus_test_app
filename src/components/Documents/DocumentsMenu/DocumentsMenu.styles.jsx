import styled, { css } from 'styled-components';
import { Box, Typography } from '@material-ui/core';
import { ArrowDropDown, ArrowRight } from '@material-ui/icons';

export const Container = styled(Box)`
  display:flex;
  flex-direction:column;
`;

export const Heading = styled(Typography)`
  font-size: 16px;
  letter-spacing: 0.23px;
  line-height:1.31;
  color: ${({ active, theme }) =>
    active ? theme.palette.primary.main : '#747782'};
  display:flex;
  align-items: center;
  font-weight:bold;
  cursor: pointer;
  border-bottom: 3px solid ${({ theme, active }) => active ? theme.palette.primary.main : 'transparent'};
  margin-bottom:17px;
`;
const iconStyles = css`
  opacity: .9;
  font-size: 20px;
`;
export const ArrowDropDownIcon = styled(ArrowDropDown)`
  ${iconStyles}
`;
export const ArrowRightIcon = styled(ArrowRight)`
  ${iconStyles}
`;

export const Text = styled(Typography)`
  font-size:16px;
  line-height: 1.31;
  letter-spacing: 0.23px;
  color: ${({ active, theme }) =>
    active ? theme.palette.primary.main : '#747782'};
  font-weight: ${({ active }) => active && 'bold'};
  border-bottom: ${({ active, theme }) =>
    active ? `3px solid ${theme.palette.primary.main}` : 'transparent'};;
  width: max-content;
  text-transform: capitalize;
`;

export const MenuItem = styled('div')`
  margin-bottom:22px;
  cursor:pointer;
  transtion:all .2s;
  
  &:hover {
    opacity: .8;
  }
`;
export const ListContainer = styled(Box)`
  padding-left:25px;
  display:flex;
  flex-direction:column;
`;
