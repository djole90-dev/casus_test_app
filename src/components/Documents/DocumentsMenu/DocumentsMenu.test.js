import React from 'react';
import { shallow } from 'enzyme';
import { ListItem, DocumentsMenu } from './DocumentsMenu';
import { Box } from '@material-ui/core'
import {
  Container,
  ListContainer
} from './DocumentsMenu.styles';

const setUp = (Component, props = {}) => {
  const component = shallow(<Component {...props} />);
  return component;
};

describe('Document Menu Component', () => {
  const baseProps = {
    categories: ['Cat1', '2', '3'],
    filterDocumentsByCategory: jest.fn()
  };

  it('should match snapshot', () => {
    const component = setUp(DocumentsMenu, baseProps);
    expect(component).toMatchSnapshot();
  });

  it('should render properly with categories', () => {
    const component = setUp(DocumentsMenu, baseProps);
    const container = component.find(Container);
    const headingWrapper = container.children().find(Box);
    const listContainer = component.find(ListContainer)
    const listItems = component.find(ListItem)

    expect(container.length).toBe(1);
    expect(headingWrapper.length).toBe(1)
    expect(listContainer.length).toBe(1)
    expect(listItems.length).toBe(3)
  });

  it('should render properly without categories', () => {
    const component = setUp(DocumentsMenu, {...baseProps, categories: []});
    const container = component.find(Container);
    const headingWrapper = container.children().find(Box);
    const listContainer = container.children().find(ListContainer)
    const listItems = listContainer.children().find(ListItem)

    expect(container.length).toBe(1);
    expect(headingWrapper.length).toBe(1)
    expect(listContainer.length).toBe(1)
    expect(listItems.length).toBe(0);
    
  });
});
