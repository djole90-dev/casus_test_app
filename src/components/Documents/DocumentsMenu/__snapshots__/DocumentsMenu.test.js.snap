// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Document Menu Component should match snapshot 1`] = `
<Styled(Styled(MuiBox))>
  <Styled(MuiBox)
    onClick={[Function]}
  >
    <Styled(WithStyles(ForwardRef(Typography)))
      active="true"
    >
      <Styled(ArrowDropDownIcon)
        onClick={[Function]}
      />
      CASUS vorlagen
    </Styled(WithStyles(ForwardRef(Typography)))>
  </Styled(MuiBox)>
  <WithStyles(ForwardRef(Collapse))
    in={true}
  >
    <Styled(Styled(MuiBox))>
      <ListItem
        active={null}
        key="Cat1"
        name="Cat1"
        onClick={[Function]}
        text="Cat1"
      />
      <ListItem
        active={null}
        key="2"
        name="2"
        onClick={[Function]}
        text="2"
      />
      <ListItem
        active={null}
        key="3"
        name="3"
        onClick={[Function]}
        text="3"
      />
    </Styled(Styled(MuiBox))>
  </WithStyles(ForwardRef(Collapse))>
</Styled(Styled(MuiBox))>
`;
